### ISSUE TITLE
opal specs not working

### ISSUE THREAD

**Issue by cantonic** [opal/opal-rails#33]
Hi there,

running `rake opal:spec` results in this:

```
WARNING: Cannot handle dynamic require -- rspec/core/configuration:
WARNING: Cannot handle dynamic require -- rspec/core/configuration:533
WARNING: Cannot handle dynamic require -- rspec/core/configuration:857
WARNING: Cannot handle dynamic require -- rspec/core/configuration:1095
WARNING: Cannot handle dynamic require -- rspec/core/configuration:1095
WARNING: Cannot handle dynamic require -- rspec/core:
WARNING: Cannot handle dynamic require -- rspec/core:200
WARNING: Cannot handle dynamic require -- rspec/core:200
ReferenceError: Can't find variable: jQuery

  http://localhost:9999/assets/opal-rspec20150107-17829-1hvhbdt.js:34806
  http://localhost:9999/assets/opal-rspec20150107-17829-1hvhbdt.js:43185
Specs timed out
```

this is the file I am trying to test:

```ruby
require "fullcalendar"
require "events_new"

class CalendarsIndex
  def initialize(collection, selector='body')
    @collection = collection
    @element = Element.find(selector)
    render_full_calendar
    add_events
  end
  
  attr_reader :element, :collection

  def render_full_calendar
    fc_options = {
      header: {
        left: 'prev,next today',
        center: 'title',
        right: 'month,basicWeek,basicDay'},
      selectable: true,
      selectHelper: true,
      select: -> (starts, ends) { show_new_form(starts, ends) }
    }

    Element.expose :fullCalendar
    element.fullCalendar(fc_options.to_n)
  end

  def add_events
    element.fullCalendar('addEventSource', collection)
  end  

  def show_new_form(starts, ends)
    event_form = EventsNew.new(starts, ends)
  end
end

Document.ready? do
  events = Native(`window.gon.events`)
  CalendarsIndex.new(events, "#fullcalendar")
end
```

I found out that `fullcalendar` is causing the problem, so I removed the require statement from the top and required `fullcalendar` in ´app/assets/javascripts/application.js.rb`. Now I get this error:

```
WARNING: Cannot handle dynamic require -- rspec/core/configuration:
WARNING: Cannot handle dynamic require -- rspec/core/configuration:533
WARNING: Cannot handle dynamic require -- rspec/core/configuration:857
WARNING: Cannot handle dynamic require -- rspec/core/configuration:1095
WARNING: Cannot handle dynamic require -- rspec/core/configuration:1095
WARNING: Cannot handle dynamic require -- rspec/core:
WARNING: Cannot handle dynamic require -- rspec/core:200
WARNING: Cannot handle dynamic require -- rspec/core:200
NameError: uninitialized constant Object::Document

  http://localhost:9999/assets/opal-rspec20150107-18079-1qg5hua.js:2804
  http://localhost:9999/assets/opal-rspec20150107-18079-1qg5hua.js:1665
  http://localhost:9999/assets/opal-rspec20150107-18079-1qg5hua.js:51
  http://localhost:9999/assets/opal-rspec20150107-18079-1qg5hua.js:34790
  http://localhost:9999/assets/opal-rspec20150107-18079-1qg5hua.js:1114 in load
  http://localhost:9999/assets/opal-rspec20150107-18079-1qg5hua.js:1136 in require
  http://localhost:9999/assets/opal-rspec20150107-18079-1qg5hua.js:2856
  http://localhost:9999/assets/opal-rspec20150107-18079-1qg5hua.js:34982
  http://localhost:9999/assets/opal-rspec20150107-18079-1qg5hua.js:34984
Specs timed out
```

Any idea what I might be missing?

**Comment by elia** [comment]
> NameError: uninitialized constant Object::Document

seems that you're missing opal-jquery, try requiring it

**Comment by cantonic** [comment]
I tried several things, but I guess this had to do with problems in the sprockets version i've been using. i am closing this for now.

**Comment by elia** [comment]
@cantonic if there's an incompatible version I'd like to fix the issue or exclude it as a gem dependency

which version were you using?

**Comment by cantonic** [comment]
@elia I was using opal-rails v0.7.0.beta2

**Comment by focused** [comment]
I use opal-rails-0.7.0, just enter `rake opal:spec`, and result is:
```
WARNING: Cannot handle dynamic require -- rspec/core/configuration:
WARNING: Cannot handle dynamic require -- rspec/core/configuration:533
WARNING: Cannot handle dynamic require -- rspec/core/configuration:857
WARNING: Cannot handle dynamic require -- rspec/core/configuration:1095
WARNING: Cannot handle dynamic require -- rspec/core/configuration:1095
WARNING: Cannot handle dynamic require -- rspec/core:
WARNING: Cannot handle dynamic require -- rspec/core:200
WARNING: Cannot handle dynamic require -- rspec/core:200
NameError: uninitialized constant Object::Document

  http://localhost:9999/assets/opal-rspec20150324-27081-10d8kis.js:3041
  http://localhost:9999/assets/opal-rspec20150324-27081-10d8kis.js:1873
  http://localhost:9999/assets/opal-rspec20150324-27081-10d8kis.js:70
  http://localhost:9999/assets/opal-rspec20150324-27081-10d8kis.js:34683
  http://localhost:9999/assets/opal-rspec20150324-27081-10d8kis.js:1292 in load
  http://localhost:9999/assets/opal-rspec20150324-27081-10d8kis.js:1314 in require
  http://localhost:9999/assets/opal-rspec20150324-27081-10d8kis.js:3094
  http://localhost:9999/assets/opal-rspec20150324-27081-10d8kis.js:34695
  http://localhost:9999/assets/opal-rspec20150324-27081-10d8kis.js:34696
```

spec-opal/example_spec.rb:
```ruby
describe 'a spec' do
  it 'has successful examples' do
    'I run'.should =~ /run/
  end
end
```

**Comment by focused** [comment]
Can i test opal files, or is it broken?  =)

**Comment by elia** [comment]
Do they work when visiting /opal_spec?

Anyway major fixes on the matter are already in master and 0.8 should be on its way

**Comment by focused** [comment]
When i visit /opal_spec i see: `0 examples, 0 failures`. I place my example spec in "spec-opal" directory inside my rails application, it seems that opal spec cannot find it.

**Comment by elia** [comment]
Right now the easiest solution is to use master or wait for 0.8

If you go with master be sure to use opal master too

**Comment by focused** [comment]
Ok, `/opal_spec` works fine if i place spec file in `assets/javascripts/spec`, but i need to add spec files to `Rails.application.config.assets.precompile`, is there any way to avoid this error message?

```
Asset filtered out and will not be served: add `Rails.application.config.assets.precompile += %w( spec/example_spec.js )` to `config/initializers/assets.rb` and restart your server
```

**Comment by focused** [comment]
With version 0.7.0 i found a solution for rails application:
```ruby
# config/initializers/assets.rb

files = Rails.env.development? ? %w(opal-rspec-runner.js spec/*.js) : %w()
files += %w(custom.js)

Rails.application.config.assets.precompile += files
```

Visiting `/opal_spec` just works with my spec files in `assets/javascripts/spec`.

**Comment by elia** [comment]
Thanks! will update opal-rails with this fix

**Comment by elia** [comment]
Should be fixed by 9e07509f
